lfric_macros/apply_macros.py

```diff
@@ -744,41 +760,55 @@ def combine_macros(self, import_order):
         for meta_import in import_order:
             meta_import = self.get_full_import_path(meta_import)
             if (
-                meta_import in self.parsed_macros
-                and self.parsed_macros[meta_import]["commands"]
+                meta_import in self.target_macros
+                and "commands" in self.target_macros[meta_import]
             ):
                 # Add a comment labelling where these commands came from
                 full_command += (
                     "        # Commands From: "
                     f"{self.parse_application_section(meta_import)}\n"
                 )
-                if self.parsed_macros[meta_import]["commands"].strip("\n"):
-                    full_command += self.parsed_macros[meta_import]["commands"] + "\n"
+                if self.target_macros[meta_import]["commands"].strip("\n"):
+                    full_command += self.target_macros[meta_import]["commands"] + "\n"
                 else:
                     full_command += "        # Blank Upgrade Macro\n"
+
+                if self.target_macros[meta_import]["author"] != "Unknown":
+                    self.author = self.target_macros[meta_import]["author"]
+                if self.target_macros[meta_import]["ticket_number"] != "TTTT":
+                    self.ticket_number = self.target_macros[meta_import][
+                        "ticket_number"
+                    ]
         return full_command
 
-    def write_new_macro(self, meta_dir, full_command):
+    def write_new_macro(self, meta_dir, full_command, macro):
         """
         Write out the new macro with all relevant commands to the versions.py
         file
         Inputs:
             - meta_dir, path to the metadata directory with a versions.py file
             - full_command, str of the combined macro commands
+            - macro, the parsed macro being written
         """
 
-        parsed_macro = self.parsed_macros[meta_dir]
         filepath = os.path.join(meta_dir, "versions.py")
         temppath = os.path.join(meta_dir, ".versions.py")
         shutil.copy(filepath, temppath)
 
+        author = macro["author"]
+        if not author:
+            author = self.author
+        ticket_number = macro["ticket_number"]
+        if not ticket_number:
+            ticket_number = self.ticket_number
```

Comment:
Maybe use a logical or?

```suggestion
        author = macro["author"] or self.author
        ticket_number = macro["ticket_number"] or self.ticket_number
```